### PR TITLE
gdb attaching to process during tests has non-sudo solution

### PR DESCRIPTION
**Pull request by instagibbs**
There are some security considerations, so a link is attached.

### CODE REVIEW

**Comment by instagibbs** [comment]
I'm not sure what the policy is about linking static web-pages for informational purposes.

File: test/README.md [hunk 1]
```diff
@@ -175,7 +175,7 @@ cat /tmp/user/1000/testo9vsdjo3/node1/regtest/bitcoind.pid
 gdb /home/example/bitcoind <pid>
 ```
 
-Note: gdb attach step may require `sudo`
+Note: gdb attach step may require ptrace_scope to be set to 0, or sudo. See this link for considerations: https://www.kernel.org/doc/Documentation/security/Yama.txt
```
Comment (laanwj) [inline test/README.md]: Please keep the `quotes`

**Comment by fanquake** [comment]
utACK f96dbd1

**Comment by laanwj** [comment]
utACK f96dbd1bbeeea82f07bc71c695fb17e8d7c9f1aa

**Pull request merged**